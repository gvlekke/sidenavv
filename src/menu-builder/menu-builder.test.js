import MenuBuilder from './menu-builder.js';

describe('MenuBuilder build', () => {
  it('should return a ul with a collection of li main elements', () => {
    const items = [
      {
        label: 'this is a label',
        link: '/',
        items: []
      },
      {
        label: 'this is a second label',
        link: '/test',
        items: []
      }
    ];

    expect(MenuBuilder.build(items)).toMatchSnapshot();
  });
});

describe('MenuBuilder buildItems', () => {
  it('should return a collection of li elements', () => {
    const items = [
      {
        label: 'this is a label',
        link: '/',
        items: []
      },
      {
        label: 'this is a second label',
        link: '/test',
        items: []
      }
    ];

    expect(MenuBuilder.buildItems(items)).toMatchSnapshot();
  });
});

describe('MenuBuilder buildItem', () => {
  it('should return one li item', () => {
    const item = {
      label: 'this is a label',
      link: '/',
      items: []
    };

    expect(MenuBuilder.buildItem(item)).toMatchSnapshot();
  });

  it('should return one li item with subitems', () => {
    const item = {
      label: 'this is a label',
      link: '/',
      items: [
        {
          label: 'this is a sub label',
          link: '/',
          items: []
        },
        {
          label: 'this is a second sub label',
          link: '/',
          items: []
        }
      ]
    };

    expect(MenuBuilder.buildItem(item)).toMatchSnapshot();
  });
});
