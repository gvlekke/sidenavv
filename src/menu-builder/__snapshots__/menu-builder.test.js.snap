// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBuilder build should return a ul with a collection of li main elements 1`] = `
"<nav class=\\"sidenavv__items\\"><ul>
      <li class=\\"sidenavv__item \\">
        <span class=\\"sidenavv__item__label\\">this is a label</span>
      
      </li>
    
      <li class=\\"sidenavv__item \\">
        <span class=\\"sidenavv__item__label\\">this is a second label</span>
      
      </li>
    </ul></nav>"
`;

exports[`MenuBuilder buildItem should return one li item 1`] = `
"
      <li class=\\"sidenavv__item \\">
        <span class=\\"sidenavv__item__label\\">this is a label</span>
      
      </li>
    "
`;

exports[`MenuBuilder buildItem should return one li item with subitems 1`] = `
"
      <li class=\\"sidenavv__item sidenavv__item--has-sub\\">
        <span class=\\"sidenavv__item__label\\">this is a label</span>
      
        <div class=\\"sidenavv__item__submenu-container\\">
          <ul>
      <li class=\\"sidenavv__item \\">
        <span class=\\"sidenavv__item__label\\">this is a sub label</span>
      
      </li>
    
      <li class=\\"sidenavv__item \\">
        <span class=\\"sidenavv__item__label\\">this is a second sub label</span>
      
      </li>
    </ul>
        </div>
      
      </li>
    "
`;

exports[`MenuBuilder buildItems should return a collection of li elements 1`] = `
"<ul>
      <li class=\\"sidenavv__item \\">
        <span class=\\"sidenavv__item__label\\">this is a label</span>
      
      </li>
    
      <li class=\\"sidenavv__item \\">
        <span class=\\"sidenavv__item__label\\">this is a second label</span>
      
      </li>
    </ul>"
`;
